src/dialog/dialog.service.ts

```diff
@@ -157,4 +162,21 @@ export class DialogService {
 			}
 		}
 	}
+
+	/**
+	 * Fix for safari hijacking clicks.
+	 *
+	 * Runs on `ngOnInit` of every dialog. Ensures we don't have multiple listeners
+	 * because having many of them could degrade performance in certain cases (and is
+	 * not necessary for our use case)
+	 *
+	 * This is an internally used function, can change at any point (even get removed)
+	 * and changes to it won't be considered a breaking change. Use at your own risk.
+	 */
+	singletonClickListen() {
```

Comment:
`singletonClickListen` works ... I can't think of anything else right now that makes more sense 👍